integration-tests/test_redaction.py

```diff
@@ -0,0 +1,136 @@
+def test_redaction_on_cmd(insights_client, tmp_path, removed_command):
+    """
+    Configure command redaction in /etc/insights-client/file-redaction.yaml,
+    so related commands should not exist in collection archive.
+    """
+    try:
+        configure_file_redaction("Command", removed_command)
+        assert not check_archive(insights_client, tmp_path, removed_command)
+    finally:
+        os.remove(FILE_REDACTION_FILE)
+
+
+@pytest.mark.parametrize("not_removed_file", TEST_FILE)
+def test_redaction_not_on_file(insights_client, tmp_path, not_removed_file):
+    """
+    Do not configure files in /etc/insights-client/file-redaction.yaml,
+    so related files should exist in collection archive.
+    """
+    assert check_archive(insights_client, tmp_path, not_removed_file)
+
+
+@pytest.mark.parametrize("removed_file", TEST_FILE)
+def test_redaction_on_file(insights_client, tmp_path, removed_file):
+    """
+    Redact files in /etc/insights-client/file-redaction.yaml,
+    so related files should not exist in collection archive.
+    """
+    try:
+        configure_file_redaction("File", removed_file)
+        assert not check_archive(insights_client, tmp_path, removed_file)
+    finally:
+        os.remove(FILE_REDACTION_FILE)
+
+
+def test_redaction_on_pattern_hostname(insights_client, tmp_path):
+    """
+    Test if the satellite hostname is obfuscated
+    Related to https://issues.redhat.com/browse/RHEL-2471
+
+    """
+
+    # Record the current hostname
+    with open("/etc/hostname", "r") as f:
+        previous_hostname = f.read().strip()
+
+    try:
+        # Set a new hostname
+        hostname_to_test = "insights-client-satellite"
+        set_hostname(hostname_to_test)
+        cmd = ["hostnamectl", "status"]
+        output = subprocess.check_output(cmd, universal_newlines=True)
+        assert hostname_to_test in output
+
+        # redaction on hostname:
+        configure_content_redaction(hostname_to_test)
+
+        # check the collection, if redaction pattern is filtered out
+        archive_name = "test_archive_" + str(uuid.uuid4()) + ".tar.gz"
+        archive_location = tmp_path / archive_name
+        insights_client.run("--register", "--output-file=%s" % archive_location)
+        with tarfile.open(archive_location, "r") as tar:
+            for w_file in tar.getmembers():
+                extracted_file = tar.extractfile(w_file)
+                if extracted_file is not None:
+                    logging.info(w_file.name)
+                    file_content = extracted_file.read().decode()
+                    assert hostname_to_test not in file_content
+
+    finally:
+        set_hostname(previous_hostname)
+        os.remove(CONTENT_REDACTION_FILE)
+
+
+def check_archive(insights_client, tmp_path, tested):
+    archive_name = "test_archive_" + str(uuid.uuid4()) + ".tar.gz"
+    archive_location = tmp_path / archive_name
+    logging.info(archive_location)
```

Comment:
nit: we can remove this logging statement from here as it does not add significant value because tmp_path erases this path after test anyways